graph/adjacency/impl/ThingAdjacencyImpl.java

```diff
@@ -48,7 +51,7 @@
     final ThingVertex owner;
     final Encoding.Direction.Adjacency direction;
     final ConcurrentMap<InfixIID.Thing, Set<InfixIID.Thing>> infixes;
-    final ConcurrentMap<InfixIID.Thing, Set<ThingEdge>> edges;
+    final ConcurrentMap<InfixIID.Thing, Map<EdgeIID.Thing, ThingEdge>> edges;
```

Comment:
@haikalpribadi  this is a major change - I'm adding an inner index by IID, so that we can look up edges and set the `isInferred` again later - a set doesn't let us retrieve the item in the set! Note that i think in terms of storage, i think this is free because a set is a map with a `boolean` as the value - now we're actually making use of it!